src/Suave/Utils/Collections.fs

```diff
@@ -21,6 +21,11 @@ let getFirst (target : NameValueList) (key : string) =
   | Some b -> Choice1Of2 b
   | None   -> Choice2Of2 (sprintf "Couldn't find key '%s' in NameValueList" key)
 
+let getAll (target : NameValueList) (key : string) =
+  match target |> List.choose (fun (a, b) -> if a.Equals key then Some b else None) with
```

Comment:
No, the headers *are* lowercased; just add a comment about that and you're done.